src/error.rs

```diff
@@ -162,3 +162,60 @@ impl From<untrusted::EndOfInput> for Error {
         Error::BadDer
     }
 }
+
+// Return the more specific of two errors.
+pub(crate) fn most_specific_error(prev: Error, new: Error) -> Error {
+    // Assign an error a numeric value ranking it by specificity.
+    #[allow(clippy::as_conversions)] // We won't exceed u32 errors.
+    fn rank(e: Error) -> u32 {
+        match e {
+            // Errors related to certificate validity
+            Error::CertNotValidYet | Error::CertExpired => 100027,
```

Comment:
If we can rely on exhaustiveness instead of having the final rule, can we use simpler numbers? Would be good to have a comment explaining why these numbers were chosen rather than something simpler.

Reply:
I switched to simpler numbers :+1: I think the intent of the larger numbers from the original was to leave space for the errors that were being ranked by the catch-all. Since we aren't doing that anymore the simpler ranks make sense to me.